Userland/Libraries/LibWeb/HTML/CloseWatcherManager.h

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright (c) 2024, the Ladybird developers.
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <AK/Forward.h>
+#include <LibWeb/DOM/EventTarget.h>
+
+namespace Web::HTML {
+
+// https://html.spec.whatwg.org/multipage/interaction.html#close-watcher-manager
+class CloseWatcherManager final : public Bindings::PlatformObject {
+    WEB_PLATFORM_OBJECT(CloseWatcherManager, Bindings::PlatformObject);
+    JS_DECLARE_ALLOCATOR(CloseWatcherManager);
+
+public:
+    [[nodiscard]] static JS::NonnullGCPtr<CloseWatcherManager> create(JS::Realm&);
+
+    void add(JS::NonnullGCPtr<CloseWatcher>);
+    void remove(CloseWatcher const&);
+
+    bool process_close_watchers();
+
+    void notify_about_user_activation();
+    bool can_prevent_close();
+
+private:
+    explicit CloseWatcherManager(JS::Realm&);
+
+    Vector<Vector<JS::NonnullGCPtr<CloseWatcher>>> m_groups;
```

Comment:
These all need to be visited by `CloseWatcherManager::visit_edges()`.

Reply:
The booleans don't need visiting right? so just Base::visit_edges and visit(m_groups) are needed?

Reply:
That's right :)